pkg/controller/replica.go

```diff
@@ -300,13 +313,34 @@ func NewReplicaController(mgr manager.Manager, controllerSharedState *SharedStat
 			return false
 		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			return true
+			objOld, okOld := e.ObjectOld.(*azdiskv1beta2.AzVolumeAttachment)
+			objNew, okNew := e.ObjectNew.(*azdiskv1beta2.AzVolumeAttachment)
+			if okOld && okNew {
+				demoteRequestUpdate := isDemotionRequested(objOld) && isDemotionRequested(objNew)
```

Comment:
Should this be `!isDemotionRequested(objOld) && isDemotionRequested(objNew)`?

Reply:
yeah, you're right : )